app/javascript/packs/userProfileSettings.js

```diff
@@ -4,10 +4,11 @@ const profileFields = document.querySelectorAll('[id^="profile["]');
 export function fieldCharacterLimits() {
   profileFields.forEach((node) => {
     const field = document.getElementById(node.id);
-    const fieldValueLength = field.value.length;
+    const fieldValueLength = [...field.value].length; // 😀 counts as 1
```

Comment:
Maybe I'm missing something but won't [...field.value].length` result in the same value as `field.value.length`?   Value is a string, so character array length or string length will result in the same length.

Reply:
I really wish JavaScript were such a well-behaved language. But sadly:
``` js
console.log("😂".length); // → 2 (👎)
console.log([..."😂"].length); // → 1 (👍)
```
Because the JS encoding model is an entire technical debt from the previous century. The `[...string]` is [more or less an idiom](https://medium.com/@tanishiking/count-the-number-of-unicode-code-points-in-javascript-on-cross-browser-62c32b8d919c) to get a correct code point-based count of a string (like how Ruby and PgSQL understand it; if this value really means something to the DB). I'm kind of surprised that they haven't provided an more intuitive way to do this as of ES2021.

Reply:
Thanks for the explanation @yheuhtozr!